src/content/init.js

```diff
@@ -146,47 +146,52 @@ async function checkLoaded(settings) {
         pre.style.display = 'none';
     }
 
-    if (!loaded) {
-        flushData(settings);
-        loadedTimer = requestAnimationFrame(() =>
-            checkLoaded(settings).catch(rollbackPageChanges)
-        );
-        return;
-    }
-
     if (pre !== null) {
+        if (!loaded) {
+            flushData(settings);
+            loadedTimer = requestAnimationFrame(() =>
+                checkLoaded(settings).catch(rollbackPageChanges)
+            );
+            return;
+        }
+
         flushData(settings);
         pushChunk(null); // end of input
 
-        const [{ initDiscovery }, json] = await Promise.all([
-            import(chrome.runtime.getURL('discovery.js')),
-            data
-        ]);
-
-        await initDiscovery({
-            node: document.body,
-            raw: Object.defineProperties({}, {
-                firstSlice: {
-                    value: totalSize < firstSliceMaxSize * 2 ? null : firstSlice
-                },
-                size: {
-                    value: totalSize
-                },
-                json: totalSize <= firstSliceMaxSize ? { value: firstSlice } : {
-                    configurable: true,
-                    get() {
-                        return Object.defineProperty(this, 'json', {
-                            value: pre.textContent
-                        }).json;
+        const json = await data;
+
+        window.__discoveryPreloader = preloader; // eslint-disable-line no-underscore-dangle
+
+        window.__discoveryOptions = [ // eslint-disable-line no-underscore-dangle
```

Comment:
I believe there is no need for exposing options to global until `postMessage` is used.

Reply:
It is needed in case of `executeScript` fallback. Pushed double init fixed version, please check.